### ISSUE TITLE
Self-contained `State` module

### ISSUE THREAD

**Issue by imalygin** [hiero-ledger/hiero-consensus-node#15574]
This is the logical continuation of the Public Merkle API initiative (#10988). Also, it overlaps with #14772.

There are two main goals of this Epic:
1. Remove all platform specifics from the State module
2. Add state lifecycle management to the State module in an generic (non-platform specific) way

Currently, a Block node would have to not only use the `swirlds-state-api` module but also `swirlds-platform-core` as a dependency, which, in turn, contains functionality that is not relevant in the context of the state (consensus, gossip, transaction handling).

We need a State module that has not only a set of API interfaces but also all the necessary implementations to represent the full Merkle state.

### Platform Dependency Removal
- [ ] https://github.com/hashgraph/hedera-services/issues/16221
- [ ] https://github.com/hashgraph/hedera-services/issues/16380
- [ ] https://github.com/hashgraph/hedera-services/issues/16193
- [ ] https://github.com/hashgraph/hedera-services/issues/15638
- [ ] https://github.com/hashgraph/hedera-services/issues/16322
- [ ] https://github.com/hashgraph/hedera-services/issues/16223
- [ ] https://github.com/hashgraph/hedera-services/issues/16436
- [ ] https://github.com/hashgraph/hedera-services/issues/16569
- [ ] https://github.com/hashgraph/hedera-services/issues/16589

### Lifecycle Management
- [ ] https://github.com/hashgraph/hedera-services/issues/15620
- [ ] https://github.com/hashgraph/hedera-services/issues/17023

**Comment by poulok** [comment]
We discussed at DevCon that this State Module may also need to include the management of states similarly to the `SwirldStateManager` which keeps a thread-safe reference to the mutable state and the latest immutable state, but more than that. It could include logic that does things like ensure that only a single loaded state is mutable and that it is always the latest one even if the user of the module loads another, older state from disk.

**Comment by thenswan** [comment]
Current plan of moving forward with this epic.

1. Remove the `State` class:
   - Issue: #15638. It should be broken down into 2 separate issues: the original should be about removing the `State` class and follow up should be about removing `MerkleRoot` interface.
   - PR: #16352
2. #16322
3. Remove `MerkleRoot` interface.
   - `SwirldState getSwirldState()` can be superseded by lifecycle methods in `MerkleStateRoot`;
   - `getReadablePlatformState`, `getWritablePlatformState` and `updatePlatformState` will be redundant after #16352, so it should be easy to simplify things;
   - `getInfoString` should be moved to the `MerkleInternal`;
   -  `State` interface also has `copy` and `createSnapshot`, so they can be simply deleted.

**Comment by imalygin** [comment]
Moved tasks related to the state lifecycle management to https://github.com/hashgraph/hedera-services/issues/17540
